src/components/changeData/changeUserInfo.tsx

```diff
@@ -0,0 +1,101 @@
+import React, { useState, useEffect } from 'react';
+import { AxiosError } from 'axios';
+import { useQuery } from '@tanstack/react-query';
+import ChangeInfoData from '../../types/ChangeInfoData';
+import ServerError from '../../types/ServerError';
+import { fetchUser } from '../../api/client';
+import useChangeUserInfo from '../../hooks/useChangeUserInfo';
+
+function ChangeData() {
+  const infoMut = useChangeUserInfo();
+
+  const [infoForm, setInfoForm] = useState<ChangeInfoData>({
+    firstName: '',
+    lastName: '',
+  });
+
+  const { data: userResponse, isLoading } = useQuery({
+    queryKey: ['me'],
+    queryFn: fetchUser,
+    retry: false,
+  });
+  const user = userResponse?.data;
+
+  useEffect(() => {
+    if (user) {
+      setInfoForm({
+        firstName: user.firstName,
+        lastName: user.lastName,
+      });
+    }
+  }, [user]);
+
+  if (isLoading) return <p>Loading...</p>;
+
+  const handleChange =
+    <F extends object>(setter: React.Dispatch<React.SetStateAction<F>>) =>
+    (e: React.ChangeEvent<HTMLInputElement>) => {
+      const { name, value } = e.target;
+      setter((prev: F) => ({ ...prev, [name]: value }) as F);
+    };
```

Comment:
to sie wydaje zbyt skomplikowane. `handleChange` powinien przyjmowac klucz, ktory modyfikujesz. cos w tym stylu:
```ts
const handleChange = (e: React.ChangeEvent<HTMLInputElement>), key: string) => {
  setInfoForm({...infoForm, key: e.target.value});
}
```

i potem w input:
```tsx
<input
  ...
  onChange={(e) => handleChange(e, 'firstName'))}
  ...
/>
```